spectree/spec.py

```diff
@@ -178,6 +194,9 @@ async def async_validate(*args, **kwargs):
                     setattr(validation, name, model_key)
 
             if resp:
+                # Make sure that the endpoint specific status code and data model for
+                # validation errors shows up in the response spec.
+                resp.add_model(validation_error_status, ValidationError, replace=False)
```

Comment:
I see. You are right. This PR doesn't change the current behavior. Sorry, I haven't read the source code for a long time. 🤣 I thought the `ValidationError` will be added to the endpoint if and only if this endpoint has at least one `BaseModel` in `(headers, cookies, query, json, resp)`.

> The only thing that changes with this PR, is where the response for the validation error is added to the Response object. Previously it was added in Response.__init__. But since the status code for validation errors can now be different for each endpoint, I think it makes sense to add it in the validate method instead.

It's OK to make these changes.